src/planar/postprocessors.jl

```diff
@@ -51,32 +51,26 @@ end
 # Replace the tensor operations created by `TO.instantiate` with the corresponding
 # planar operations, immediately inserting them with `GlobalRef`.
 
-# NOTE: work around a somewhat unfortunate interface choice in TensorOperations, which we will correct in the future.
-_planaradd!(C, p, A, α, β, backend...) = planaradd!(C, A, p, α, β, backend...)
-_planartrace!(C, p, A, q, α, β, backend...) = planartrace!(C, A, p, q, α, β, backend...)
-function _planarcontract!(C, pAB, A, pA, B, pB, α, β, backend...)
-    return planarcontract!(C, A, pA, B, pB, pAB, α, β, backend...)
-end
 # TODO: replace _planarmethod with planarmethod in everything below
-const _PLANAR_OPERATIONS = (:_planaradd!, :_planartrace!, :_planarcontract!)
+const _PLANAR_OPERATIONS = (:planaradd!, :planartrace!, :planarcontract!)
 
 function _insert_planar_operations(ex)
     if isexpr(ex, :call)
         if ex.args[1] == GlobalRef(TensorOperations, :tensoradd!)
             conjA = popat!(ex.args, 5)
-            @assert conjA == :(:N) "conj flag should be `:N` ($conjA)"
-            return Expr(ex.head, GlobalRef(TensorKit, Symbol(:_planaradd!)),
+            @assert !conjA "conj flag should be disabled"
+            return Expr(ex.head, GlobalRef(TensorKit, Symbol(:planaradd!)),
                         map(_insert_planar_operations, ex.args[2:end])...)
         elseif ex.args[1] == GlobalRef(TensorOperations, :tensorcontract!)
-            conjB = popat!(ex.args, 9)
-            conjA = popat!(ex.args, 6)
-            @assert conjA == conjB == :(:N) "conj flag should be `:N` ($conjA), ($conjB)"
-            return Expr(ex.head, GlobalRef(TensorKit, Symbol(:_planarcontract!)),
+            conjB = popat!(ex.args, 8)
+            conjA = popat!(ex.args, 5)
+            @assert !conjA && !conjB "conj flags should be disabled ($conjA), ($conjB)"
+            return Expr(ex.head, GlobalRef(TensorKit, Symbol(:planarcontract!)),
                         map(_insert_planar_operations, ex.args[2:end])...)
         elseif ex.args[1] == GlobalRef(TensorOperations, :tensortrace!)
             conjA = popat!(ex.args, 6)
-            @assert conjA == :(:N) "conj flag should be `:N` ($conjA)"
-            return Expr(ex.head, GlobalRef(TensorKit, Symbol(:_planartrace!)),
+            @assert !conjA "conj flag should be disabled"
+            return Expr(ex.head, GlobalRef(TensorKit, Symbol(:planartrace!)),
                         map(_insert_planar_operations, ex.args[2:end])...)
         elseif ex.args[1] in TensorOperations.tensoroperationsfunctions
```

Comment:
I am confused by this. Why do `tensoradd!` & friends are already of the `GlobalRef` form (i.e. we check `ex.args[1] == GlobalRef(TensorOperations, :tensoradd!)` but would the other functions in `tensoroperationsfunction` not yet have the `GlobalRef` asssignment, and need to have it inserted by this postprocessor? It's old code, so it's probably a question to myself 😄 .

Reply:
The other functions in `tensoroperations` do already have this, except for the ones that get added by different postprocessors (eg `free_temporaries`). In principle, the postprocessor that adds the `GlobalRef`s should just come last, but it is already part of the `TensorParser()` call so it's a bit inelegant :upside_down_face:

Reply:
Oh right, I see, postprocessor steps that the planar parser adds after the `addtensoroperations` postprocessor that is by default present in `TensorParser`.